reify.go

```diff
@@ -343,7 +343,9 @@ func reifyGetField(
 		return err
 	}
 
-	to.Set(pointerize(to.Type(), v.Type(), v))
+	if v.IsValid() {
+		to.Set(pointerize(to.Type(), v.Type(), v))
+	}
```

Comment:
Given the test case I don't see why `v` can be invalid. Do we have an original issue with sample code? `v` should always be assignable to `to`, and return a zero value at least.

As reifyGetField is an internal function processing a sub-tree only, I wonder if the issue can occur in other edge cases as well.

Reply:
> Do we have an original issue with sample code?

This is the exact extracted part from the linked PR in the issue. Just unpack the given YAML file into a structure with `interface{}`

Reply:
What I'm wondering about is, if this or a similar error can occur when we unpack into some alternative type. For example if one unpacks into a `map`, then the call `to.SetMapIndex(key, v)` at line 224 would be the equivalent for maps. For arrays/slices the equivalent call is `to.Index(idx).Set(v)` in `reifyDoArray`. Both calls do not check for an invalid `reflect.Value` yet. Given the bug at hand, I doubt the other 2 cases have tests that might show an error.

Reply:
I've added more test cases, but couldn't reproduce. If you see any potential weird structure that can cause an error, I can add a test case.

Other way round is to wrap all `to.Set()` calls with a condition checking if the target is valid.

Reply:
If the tests show that we are fine, I'm ok with that.

maybe we can add these combinations for maps as well:
```
map[string]interface{}{}  , `{"a": []}`
map[string]interface{}{"a": nil}  , `{"a": []}`
map[string][]interface{}{}, `{"a": []}`
map[string][]interface{}{"a": nil}, `{"a": []}`
```

and for arrays:

```
[]interface{} , `[[]]`
[][]interface{}, `[[]]`
```

Reply:
I've updated test cases, but I think we need to review/accept the final unmarshaled forms (I'm referring specifically to this case: `map[string]interface{}{"a": nil}` where the field disappears).

Reply:
Hm, the field `a` should not just disappear. If the input configuration does not contain the field, or if it is an empty array, then it should not be overwritten. We want to merge the configuration into the object given. Yet one can discuss what it means to merge an empty array... But it kinda looks like we just found another bug.
I don't want increase the scope for this PR any further. How about skipping this tests that do remove `a` and create a follow up issue?

Reply:
I think I've just fixed this case by checking if the target is valid:

```go
if v.IsValid() {
    to.SetMapIndex(key, v)
}
```